common/remote/rmtssh.cpp

```diff
@@ -416,38 +444,54 @@ class CFRunSSH: public CInterface, implements IFRunSSH
                 expandCmd(cmdline,i,treefrom);
                 cmdline.append('"');
             }
-            if (dryrun) 
+            if (dryrun)
                 printf("%s\n",cmdline.str());
-            else {
+            else
+            {
+                DBGLOG(-1, "%s\n",cmdline.str());
                 Owned<IPipeProcess> pipe = createPipeProcess();
                 if (pipe->run((verbose&&!usepssh)?"FRUNSSH":NULL,cmdline.str(),workdir,
-                    useplink, // for some reason plink needs input handle
-                    true,true)) {
+                              useplink, // for some reason plink needs input handle
+                              true,true))
+                {
                     byte buf[4096];
-                    for (;;) {
+                    for (;;)
+                    {
                         size32_t read = pipe->read(sizeof(buf),buf);
                         if (!read)
                             break;
                         outbuf.append(read,(const char *)buf);
                     }
                     retcode = pipe->wait();
                     bool firsterr=true;
-                    for (;;) {
+                    for (;;)
+                    {
                         size32_t read = pipe->readError(sizeof(buf),buf);
                         if (!read)
                             break;
-                        if (firsterr) {
+
+                        if (firsterr)
+                        {
                             firsterr = false;
+                            if ((strstr((const char *)buf, "ftslave") != nullptr) && !retcode)
```

Comment:
This can't be right, hard coding any specific program name here in this general purpose class, doesn't look right.

Reply:
Referring back to previous review comments by Mark and I:
https://github.com/hpcc-systems/HPCC-Platform/pull/10335#issuecomment-322842196
and
https://github.com/hpcc-systems/HPCC-Platform/pull/10335#discussion_r133524589

As it says those comments, I don't think you can assume output on stderr = failure.
You can only assume that if the program returned an error code or it failed to start.